pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -490,6 +493,19 @@ public static void validateIngestionConfig(TableConfig tableConfig, @Nullable Sc
           Preconditions.checkState(new HashSet<>(schema.getMetricNames()).equals(aggregationColumns),
               "all metric columns must be aggregated");
         }
+
+        // This is required by MutableSegmentImpl.enableMetricsAggregationIfPossible().
+        // That code will disable ingestion aggregation if all metrics aren't noDictionaryColumns.
+        // But if you do that after the table is already created, all future aggregations will
+        // just be the default value.
+        Map<String, FieldIndexConfigs> fieldIndexConfigsMap = FieldIndexConfigsUtil.createIndexConfigsByColName(
```

Comment:
@gortiz Is there a way to only extract the `DictionaryIndexConfig` for all columns? Here we just need to find all the no-dictionary columns.

Reply:
As far as I know there is none. But in this case we can directly call `StandardIndexes.dictionary().getConfig(tableConfig, schema)`.

Specifically we could change this code to do something like:

```java
       Map<String, DictionaryIndexConfig> configPerCol = StandardIndexes.dictionary().getConfig(tableConfig, schema);
        aggregationColumns.forEach(column -> {
          DictionaryIndexConfig dictConfig = configPerCol.get(column);
          Preconditions.checkState(dictConfig != null && dictConfig.isEnabled(),
            "Aggregated column: %s must be a no-dictionary column", column);
          });
        }
```

`FieldIndexConfigsUtil.createIndexConfigsByColName` is designed to be used when we need to calculate the indexes for all columns and it includes a deserializer in order to optionally alter the way index configs are read (which is necessary in order to use IndexLoadingConfig in tests)

Reply:
great suggestion, thank you. I think is should be `.isDisabled()`. Or at least that's what tests say.

There's honestly a lot of places we might want to clean this up. The invertedIndex check is based solely on the `noDictionaryColumn` config for example. I'm surprised more people aren't running into this.